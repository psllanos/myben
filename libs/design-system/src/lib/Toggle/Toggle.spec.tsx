/* eslint-disable @typescript-eslint/no-empty-function */
import { fireEvent, render, screen } from '@testing-library/react'
import { Toggle } from '.'

describe('Toggle', () => {
    describe('when rendered with a screenreader label', () => {
        it('should display the label to screenreaders', () => {
            const component = render(<Toggle onChange={() => {}} screenReaderLabel="SRLabel" />)

            expect(screen.getByText('SRLabel')).toBeInTheDocument()
            expect(component).toMatchSnapshot()
        })
    })

    describe('when pressed', () => {
        it('should call the `onClick` callback', () => {
            const onClickMock = jest.fn()
            render(<Toggle screenReaderLabel="SRLabel" onChange={onClickMock} />)

            fireEvent.click(screen.getByRole('switch'))
            expect(onClickMock).toHaveBeenCalledTimes(1)
        })
    })

    describe('when toggle is checked', () => {
        it('should have the toggle disc moved to the right', () => {
            const component = render(
                <Toggle screenReaderLabel="SRLabel" onChange={() => {}} checked />
            )

            expect(component).toMatchSnapshot()
        })
    })

    describe('when passing a className', () => {
        it('should merge them and show all classNames', () => {
            render(
                <Toggle
                    screenReaderLabel="SRLabel"
                    onChange={() => {}}
                    className="test-classname"
                />
            )

            expect(screen.getByRole('switch')).toHaveClass('test-classname')
            // className cursor-pointer exists on the toggle by default
            expect(screen.getByRole('switch')).toHaveClass('cursor-pointer')
        })
    })
})
