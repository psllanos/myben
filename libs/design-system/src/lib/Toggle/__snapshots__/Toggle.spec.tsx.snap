// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle when rendered with a screenreader label should display the label to screenreaders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        aria-checked="false"
        class="bg-gray-600 focus:ring-gray relative inline-flex shrink-0 border-2 border-transparent rounded-full cursor-pointer transition-colors ease-in-out duration-200 focus:ring-2 focus:outline-none focus:ring-opacity-60 p-0.5 h-8 w-14"
        data-headlessui-state=""
        id="headlessui-switch-:r0:"
        role="switch"
        tabindex="0"
        type="button"
      >
        <span
          class="sr-only"
        >
          SRLabel
        </span>
        <span
          class="translate-x-0 bg-gray-200 pointer-events-none inline-block rounded-full shadow transform ring-0 transition ease-in-out duration-200 h-6 w-6"
        />
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      aria-checked="false"
      class="bg-gray-600 focus:ring-gray relative inline-flex shrink-0 border-2 border-transparent rounded-full cursor-pointer transition-colors ease-in-out duration-200 focus:ring-2 focus:outline-none focus:ring-opacity-60 p-0.5 h-8 w-14"
      data-headlessui-state=""
      id="headlessui-switch-:r0:"
      role="switch"
      tabindex="0"
      type="button"
    >
      <span
        class="sr-only"
      >
        SRLabel
      </span>
      <span
        class="translate-x-0 bg-gray-200 pointer-events-none inline-block rounded-full shadow transform ring-0 transition ease-in-out duration-200 h-6 w-6"
      />
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Toggle when toggle is checked should have the toggle disc moved to the right 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        aria-checked="true"
        class="bg-cyan focus:ring-cyan relative inline-flex shrink-0 border-2 border-transparent rounded-full cursor-pointer transition-colors ease-in-out duration-200 focus:ring-2 focus:outline-none focus:ring-opacity-60 p-0.5 h-8 w-14"
        data-headlessui-state="checked"
        id="headlessui-switch-:r2:"
        role="switch"
        tabindex="0"
        type="button"
      >
        <span
          class="sr-only"
        >
          SRLabel
        </span>
        <span
          class="translate-x-6 bg-white pointer-events-none inline-block rounded-full shadow transform ring-0 transition ease-in-out duration-200 h-6 w-6"
        />
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      aria-checked="true"
      class="bg-cyan focus:ring-cyan relative inline-flex shrink-0 border-2 border-transparent rounded-full cursor-pointer transition-colors ease-in-out duration-200 focus:ring-2 focus:outline-none focus:ring-opacity-60 p-0.5 h-8 w-14"
      data-headlessui-state="checked"
      id="headlessui-switch-:r2:"
      role="switch"
      tabindex="0"
      type="button"
    >
      <span
        class="sr-only"
      >
        SRLabel
      </span>
      <span
        class="translate-x-6 bg-white pointer-events-none inline-block rounded-full shadow transform ring-0 transition ease-in-out duration-200 h-6 w-6"
      />
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
